pkg/divert/istio/divert.go

```diff
@@ -0,0 +1,212 @@
+				return err
+			}
+
+			if !isInDivertVirtualServices(name, d.Manifest.Deploy.Divert.VirtualServices) {
+				continue
+			}
+			if err := d.retryTranslateDeveloperVirtualService(ctx, name); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func (d *Driver) Destroy(ctx context.Context) error {
+	for name := range d.cache.divertVirtualServices {
+		select {
+		case <-ctx.Done():
+			oktetoLog.Infof("destroyDivert context cancelled")
+			return ctx.Err()
+		default:
+			oktetoLog.Spinner(fmt.Sprintf("Destroying divert un virtual service %s/%s ...", d.Manifest.Deploy.Divert.Namespace, name))
+			if err := d.retryRestoreDivertVirtualService(ctx, name); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (d *Driver) GetDivertNamespace() string {
+	if d.Manifest.Deploy.Divert.Namespace == d.Manifest.Namespace {
+		return ""
+	}
+	return d.Manifest.Deploy.Divert.Namespace
+}
+
+func (d *Driver) InitCache(ctx context.Context) error {
+	d.cache = &divertIstioCache{
+		divertVirtualServices:    map[string]*istioV1beta1.VirtualService{},
+		developerVirtualServices: map[string]*istioV1beta1.VirtualService{},
+	}
+	vsList, err := virtualservices.List(ctx, d.Manifest.Deploy.Divert.Namespace, d.IstioClient)
+	if err != nil {
+		return err
+	}
+	for i := range vsList {
+		d.cache.divertVirtualServices[vsList[i].Name] = vsList[i]
+	}
+
+	vsList, err = virtualservices.List(ctx, d.Manifest.Namespace, d.IstioClient)
+	if err != nil {
+		return err
+	}
+	for i := range vsList {
+		d.cache.developerVirtualServices[vsList[i].Name] = vsList[i]
+	}
+	return nil
+}
+
+func (d *Driver) retryTranslateDeveloperVirtualService(ctx context.Context, name string) error {
+	retries := 0
+	var err error
+	for retries < UPDATE_CONFLICT_RETRIES {
+		if _, ok := d.cache.developerVirtualServices[name]; !ok {
+			vs := createIntoDeveloperVirtualService(d.Manifest, d.cache.divertVirtualServices[name])
+			err = virtualservices.Create(ctx, vs, d.IstioClient)
+			if err == nil {
+				d.cache.developerVirtualServices[name] = vs
+				return nil
+			}
+			if !k8sErrors.IsAlreadyExists(err) {
+				return err
+			}
+			return nil
+		}
+
+		vs := translateDeveloperVirtualService(d.Manifest, d.cache.developerVirtualServices[name])
+		if vs.Annotations[model.OktetoAutoCreateAnnotation] != "" {
+			vs = createIntoDeveloperVirtualService(d.Manifest, d.cache.divertVirtualServices[name])
+			vs.ResourceVersion = d.cache.developerVirtualServices[name].ResourceVersion
+		}
+		if isEqualVirtualService(vs, d.cache.developerVirtualServices[name]) {
+			return nil
+		}
+
+		err = virtualservices.Update(ctx, vs, d.IstioClient)
+		if err == nil {
+			d.cache.developerVirtualServices[name] = vs
+			return nil
+		}
+		if !k8sErrors.IsConflict(err) {
+			return err
+		}
+		retries++
+	}
+	return err
+}
+
+func (d *Driver) retryTranslateDivertVirtualService(ctx context.Context, name string) error {
+	retries := 0
```

Comment:
Nit: we could add the `retries` variable to the for loop so we can remove the `retries++`

Reply:
updated!